playbooks/roles/ocp-quay/tasks/main.yaml

```diff
@@ -0,0 +1,222 @@
+  set_fact:
+    worker: "{{ worker_list.stdout_lines }}"
+
+- name: Check if LSO (Local Storage Operator) is already installed
+  shell: |
+    oc get csv -n openshift-local-storage -o json | jq -r '.items[] | select(.metadata.name | test("local-storage-operator")) | .status.phase'
+  register: lso_csv_status
+  changed_when: false
+  failed_when: false
+
+- name: Set flag if LSO is installed
+  set_fact:
+    lso_installed: "{{ lso_csv_status.stdout == 'Succeeded' }}"
+
+# Install LSO
+- name: Install LSO - Local Storage operator
+  include_role:
+    name: ocp-lso
+  when: not lso_installed
+  vars:
+    lso_catalogsource_image: "{{ lso_index }}"
+    upi_cluster: "{{ cluster_upi }}"
+    device_path: "{{ volume_path }}" 
+    lso_namespace: "openshift-local-storage"
+    lso_channel: "stable"
+    lso_catalogsource_name: "my-operator-catalog"
+    busybox_image: "quay.io/powercloud/busybox:ubi"
+
+- name: Check if ODF (OpenShift Data Foundation) is already installed
+  shell: |
+    oc get csv -n openshift-storage -o json | jq -r '.items[] | select(.metadata.name | test("odf-operator")) | .status.phase'
+  register: odf_csv_status
+  changed_when: false
+  failed_when: false
+
+- name: Set flag if ODF is installed
+  set_fact:
+    odf_installed: "{{ odf_csv_status.stdout == 'Succeeded' }}"
+
+# ODF installation
+- name: Install ODF - Openshift Data Foundation
+  include_role:
+    name: ocp-odf-operator
+  when: not odf_installed
+  vars:
+    odf_catalogsource_image: "{{ odf_index }}"
+    update_channel: "{{ odf_channel }}"
+    test_pod_image: "quay.io/powercloud/nginx-unprivileged:latest"
+
+# Creating Project for Quay
+- name: Create quay-registry project
+  k8s:
+    state: present
+    definition:
+      apiVersion: v1
+      kind: Namespace
+      metadata:
+        name: "{{ quay_registry_namespace }}"
+        annotations:
+          openshift.io/node-selector: 'node-role.kubernetes.io/infra='
+          openshift.io/description: "Red Hat Quay Enterprise Container Image Repository"
+          openshift.io/display-name: "Quay"
+          scheduler.alpha.kubernetes.io/defaultTolerations: >-
+            [{"operator": "Exists", "effect": "NoSchedule", "key": "node-role.kubernetes.io/infra"}]
+
+- name: Switch to quay-registry project
+  command: oc project {{ quay_registry_namespace }}
+
+# Custom ImageContentSourcePolicy and CatalogSource
+- name: Create ImageContentSourcePolicy and CatalogSource
+  block:
+  - name: Include the global-secret-update role
+    include_role:
+      name: global-secret-update
+
+  - name: Include role to create ImageContentSourcePolicy and CatalogSource
+    include_role:
+      name: set-custom-catalogsource
+    vars:
+      custom_catalogsource_name: "{{ quay_catalogsource_name }}"
+      custom_catalogsource_display_name: "Custom Quay CatalogSource"
+      custom_catalogsource_image: "{{ quay_catalogsource_image }}"
+  when: quay_catalogsource_image != '' and quay_catalogsource_image != None
```

Comment:
You can extend the condition to check if var is defined along with empty string to avoid var undefined error during the specific role run.